bluesky/simulators.py

```diff
@@ -114,3 +121,401 @@ def check_limits(plan):
                 raise LimitsExceeded("This plan would put {} at {} "
                                      "which is outside of its limits, {}."
                                      "".format(msg.obj.name, pos, (low, high)))
+
+
+
+
+
+"""
+ATTRIBUTE PARAMETERS
+--------------------
+_run_start_cmds: list
+    A list of message 'commands' that are found at the start of a run.
+_run_end_cmds: list
+    A list of message 'commands' that are found at the end of a run.
+_group_start_cmds: list
+    A list of message 'commands' that are found at the begining of a 'group' (where more than
+    one command is processed in parallel).
+_group_end_cmds: list
+    A list of message 'commands' that are found at the begining of a 'group' (where more than
+    one command is processed in parallel).
+_self._timed_cmds: list.
+    A list of message 'commands' that do not start/end a run/group/flyer but are stil timed.
+__flyer_start_cmds: list.
+    A list of message 'commands' that start a 'flyer'.
+_plan_history: dict.
+    A dictionary containing information on values updated during the plan. It has key:arg 
+    pairs with keys relating to message components where each arg is a dictionary 
+    containing devices (detectors, axes) that have been updated, as keywords and a value 
+    indicating the update status. For 'set' the update status is the latest 'position' that
+    it is to be set and for 'trigger' the update status is the number of times since the 
+    beginning of the scan or since an 'unstage' event that the device has been triggered.
+
+"""
+#The commands that start/end a 'run'.
+_run_start_cmds = ['open_run']
+_run_end_cmds = ['close_run']
+#The commands that start/end a 'group', but not flyer commands.
+_group_start_cmds = ['set','trigger']   
+_group_end_cmds = ['wait']
+#The commands that don't start/end a 'group/plan' but are still timed.
+_timed_cmds = ['stage', 'unstage', 'sleep']
+#Commands that are 
+_flyer_start_cmds = ['kickoff']
+#create a reference to a dictionary called plan_history to capture set changes during the 
+#plan for later modification.
+_plan_history = {'set':{}, 'trigger':{}}
+
+def est_time(plan, print_output = True):
+    '''The call method.
+
+    This method takes in a plan, and an optional print_output kwarg, and returns an estimate for 
+    the time to complete for the plan, in addition it returns an time estimate for any subplans. 
+
+    PARAMETERS
+    ----------
+    plan : generator.
+        The bluesky plan that the est_time is to be estimated for.
+    print_output : boolean, optional.
+        Indicates if the return values should also be printed to the command line in a human 
+        readable way, default value is 'True'.
+
+    RETURN PARAMETERS
```

Comment:
Sphinx expects this to be spelled `Returns`

Reply:
fixed